src/ralph/models/selector.py

```diff
@@ -0,0 +1,154 @@
+"""Model selector definition"""
+
+from collections import Counter
+from dataclasses import dataclass
+from importlib import import_module
+from inspect import getmembers, isclass
+from itertools import chain
+from types import ModuleType
+from typing import Any, Type, Union
+
+from pydantic import BaseModel
+
+from ralph.defaults import MODEL_PATH_SEPARATOR
+from ralph.exceptions import ModelRulesException
+from ralph.utils import get_dict_value_from_path
+
+
+@dataclass(frozen=True)
+class LazyModelField:
+    """Represents a model field."""
+
+    path: tuple[str]
+
+    def __init__(self, path: str):
+        object.__setattr__(self, "path", tuple(path.split(MODEL_PATH_SEPARATOR)))
+
+
+@dataclass(frozen=True)
+class Rule:
+    """Represents a rule used for selection."""
+
+    field: LazyModelField
+    value: Union[LazyModelField, Any]
+
+    def check(self, event):
+        """Checks if event matches the rule.
+
+        Args:
+            event (dict): The event to check.
+        """
+
+        event_value = get_dict_value_from_path(event, self.field.path)
+        expected_value = self.value
+        if isinstance(expected_value, LazyModelField):
+            expected_value = get_dict_value_from_path(event, expected_value.path)
+        return event_value == expected_value
+
+
+class ModelRules(dict):
+    """Stores the list of rules for each model."""
+
+    def __setitem__(self, new_model: Type[BaseModel], new_rules: list[Rule]):
+        """Inserts a new model with it's associated list of rules.
+
+        Raises:
+            ModelRulesException: When new_rules list is a subset or superset of another rules list.
+        """
+
+        rules_set = set(new_rules)
+        for model, rules in self.items():
+            if rules_set.issubset(rules) or rules_set.issuperset(rules):
+                raise ModelRulesException(
+                    f"{new_model.__name__} rules are a subset or superset "
+                    f"of {model.__name__} rules"
+                )
+        super().__setitem__(new_model, new_rules)
```

Comment:
As the number of supported events/models grows, we should pay attention on this as it does not scale well... I mean not for now, but in a near future.

Reply:
You are right, yes indeed this will not scale well in future